src/adapter/src/coord/cluster_scheduling.rs

```diff
@@ -0,0 +1,169 @@
+                            .collect_vec();
+
+                        // Check whether there is at least one REFRESH MV needing a refresh.
+                        let local_read_ts = self.get_local_read_ts().await;
+                        decisions.push((
+                            cluster.id,
+                            refresh_mvs.into_iter().any(|mv| {
+                                let mv_frontier = &self
+                                    .controller
+                                    .storage
+                                    .collection(*mv)
+                                    .expect("the storage controller should know about MVs that exist in the catalog")
+                                    .write_frontier;
+                                mv_frontier.less_than(&local_read_ts)
+                            }),
+                        ));
+                    }
+                }
+            }
+        }
+        decisions
+    }
+
+    #[mz_ore::instrument(level = "debug")]
+    pub(crate) async fn handle_scheduling_decisions(
+        &mut self,
+        decisions: Vec<(&'static str, Vec<(ClusterId, bool)>)>,
+    ) {
+        for (policy_name, decisions) in decisions.iter() {
+            for (cluster_id, decision) in decisions {
+                self.scheduling_decisions
+                    .entry(*cluster_id)
+                    .or_insert_with(Default::default)
+                    .insert(policy_name, *decision);
+            }
+        }
+
+        // Clean up those clusters from `scheduling_decisions` that
+        // - have been dropped, or
+        // - were switched to unmanaged, or
+        // - were switched to `SCHEDULE = MANUAL`.
+        for cluster_id in self.scheduling_decisions.keys().cloned().collect_vec() {
+            match self.get_managed_cluster_config(cluster_id) {
+                None => {
+                    // Cluster have been dropped or switched to unmanaged.
+                    self.scheduling_decisions.remove(&cluster_id);
+                }
+                Some(managed_config) => {
+                    if matches!(managed_config.schedule, ClusterScheduleOptionValue::Manual) {
+                        self.scheduling_decisions.remove(&cluster_id);
+                    }
+                }
+            }
+        }
+
+        for (cluster_id, decisions) in self.scheduling_decisions.clone() {
+            // If all policies have made a decision about this cluster
+            if POLICIES.iter().all(|policy| decisions.contains_key(policy)) {
+                // check whether the cluster's state matches the needed state.
+                let needs_replica = decisions.values().contains(&true);
+                let cluster_config = self
+                    .get_managed_cluster_config(cluster_id)
+                    .expect("cleaned up non-existing and unmanaged clusters above");
+                let has_replica = cluster_config.replication_factor > 0; // Is it on?
```

Comment:
If `SCHEDULE = ON REFRESH` is going to be in charge of managing the replication factor, it seems to me like we should prohibit the user from specifying the `REPLICATION FACTOR` option when `SCHEDULE = ON REFRESH` is active. Otherwise they might create the cluster with a replication factor of something other than 1 and get confused when it doesn't stick.

Reply:
Done
416a700e3c7a01e1c76d8d00c74e718aea05e139